modules/rest_api/controllers/services/rest.php

```diff
@@ -2430,11 +2430,11 @@ private function authenticateUsingJwtUser() {
       // this scope has been claimed by the token. We allow scope as a standard
       // claim or custom claim and as an array or space separated string for
       // wider compatibility with auth servers.
-      if (isset($payloadValues['http://indicia.org.uk/scope']) && !isset($payloadValues['scope'])) {
-        $payloadValues['scope'] = $payloadValues['http://indicia.org.uk/scope'];
+      if (isset($payloadValues['http://indicia.org.uk/scope']) && !isset($payloadValues['scopes'])) {
```

Comment:
I'm not sure about this change - scope is referred to in the API self-documentation and the CHANGELOG.md file, so changing it to scopes may have wider repercussions. Is there an actual problem using scope?

Reply:
This is an example JWT payload that I captured.
```
$payloadValues
array(13)
  aud: "2977a900-8c1e-4790-b940-0d027f8d9803"
  jti: "be715920a5bad87c3c1ab17c5ebbea392dd197c9e8ca3fd905a773169b79d47b59d29eec5dde716f"
  iat: 1637055814
  nbf: 1637055814
  exp: 1637155814
  sub: "3212"
  scopes: array(1)
    0: "authenticated"
  iss: "https://butterfly-monitoring.net"
  email: "jame2@ceh.ac.uk"
  email_verified: true
  http://indicia.org.uk/user:id: "2"
  field_first_name: "Jim"
  field_last_name: "Bacon"
```
The key is `scopes` and derives from https://git.drupalcode.org/project/simple_oauth/-/blob/8.x-4.x/src/Entities/AccessTokenEntity.php#L37. 

Our code is incompatible with the Drupal code which is why I propose this change. Perhaps another generator of tokens would do differently. Should I rewrite this to make it compatible with both `scope` and `scopes`? Do we know if anyone is using the API and generating JWTs by a different means?

Reply:
I think we chose scope so as to not clash with the oAuth scopes claim, since we need it for a specific purpose (to declare the sharing mode for website agreements). So our scope contains things like "reporting" - the example above has "authenticated" which would not be recognised. 

In the JWT above, there is already a custom claim for http://indicia.org.uk/user:id, so the idea would be to add a custom claim http://indicia.org.uk/scope if a non-default sharing mode is required.

Reply:
Thanks. I was just digging a bit more and I think the scopes injected by the Drupal module are Drupal roles. I'd misunderstood the intention and will revert this change.